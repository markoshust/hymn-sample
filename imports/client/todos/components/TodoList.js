import React, { PropTypes } from 'react';
import Todo from './Todo';

const TodoList = ({
  todos,
  onTodoClick,
}) => (
  <ul>
    {todos.map(todo => (
      <Todo
        completed={todo.completed}
        key={todo.id}
        id={todo.id}
        onClick={onTodoClick}
        text={todo.text}
      />
    ))}
  </ul>
);

TodoList.propTypes = {
  todos: PropTypes.array,
  onTodoClick: PropTypes.func.isRequired,
};

export default TodoList;
