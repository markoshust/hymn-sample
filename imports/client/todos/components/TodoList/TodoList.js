import React, { PropTypes } from 'react';
import ListItem from 'client/core/components/ListItem';

const TodoList = ({
  todos,
  onTodoClick,
}) => (
  <ul>
    {todos.map(todo => (
      <ListItem
        isStriked={todo.completed}
        key={todo.id}
        id={todo.id}
        onClick={onTodoClick}
      >
        {todo.text}
      </ListItem>
    ))}
  </ul>
);

TodoList.propTypes = {
  todos: PropTypes.arrayOf(
    PropTypes.shape({
      completed: PropTypes.bool,
      id: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
    })
  ).isRequired,
  onTodoClick: PropTypes.func.isRequired,
};

export default TodoList;
