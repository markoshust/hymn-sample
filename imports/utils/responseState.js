const responseState = (response, ...state) => (
  response[state] ? response[state]() : state
);

export default responseState;
