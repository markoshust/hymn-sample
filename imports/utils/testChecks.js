import renderer from 'react-test-renderer';

const componentCheck = (component) => {
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

const renderCheck = (children) => {
  const component = renderer.create(children);

  return componentCheck(component);
};

export {
  componentCheck,
  renderCheck,
};
