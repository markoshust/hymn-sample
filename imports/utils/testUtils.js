import renderer from 'react-test-renderer';

const snap = (children) => {
  const component = renderer.create(children);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

export {
  snap, // eslint-disable-line import/prefer-default-export
};
