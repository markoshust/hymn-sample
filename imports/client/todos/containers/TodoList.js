import { connect } from 'react-redux';
import { toggle } from '../actions';
import TodoList from '../components/TodoList';

const getVisibleTodos = (todos, state) => {
  switch (state) {
    case 'SHOW_ALL':
      return todos;
    case 'SHOW_COMPLETED':
      return todos.filter(t => t.completed);
    case 'SHOW_ACTIVE':
      return todos.filter(t => !t.completed);
    default:
      throw new Error(`Unknown filter: ${state}.`);
  }
};

const mapStateToProps = state => ({
  todos: getVisibleTodos(state.todos, state.visibilityFilter),
});

const mapDispatchToProps = dispatch => ({
  onTodoClick: id => dispatch(toggle(id)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TodoList);
