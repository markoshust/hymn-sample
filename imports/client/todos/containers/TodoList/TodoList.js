import { connect } from 'react-redux';
import { todosToggle } from '../../actions';
import TodoList from '../../components/TodoList';
import { getVisibleTodos } from '../../reducers';

const mapStateToProps = state => ({
  todos: getVisibleTodos(state.todos, state.visibilityFilter),
});

const mapDispatchToProps = dispatch => ({
  onTodoClick: id => dispatch(todosToggle(id)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TodoList);
