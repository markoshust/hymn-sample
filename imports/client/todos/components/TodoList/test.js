import React from 'react';
import { renderCheck } from 'utils/testChecks';
import TodoList from './TodoList';

const onSubmit = jest.fn();

describe('<TodoList />', () => {
  it('renders no todos', () => {
    const todos = [];

    renderCheck(
      <TodoList
        onTodoClick={onSubmit}
        todos={todos}
      />
    );
  });

  it('renders with todos', () => {
    const todos = [{
      completed: false,
      id: 'a',
      text: 'item a',
    }, {
      completed: true,
      id: 'b',
      text: 'item b',
    }];

    renderCheck(
      <TodoList
        onTodoClick={onSubmit}
        todos={todos}
      />
    );
  });
});
