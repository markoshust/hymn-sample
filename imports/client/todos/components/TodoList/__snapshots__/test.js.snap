exports[`<TodoList /> renders no todos 1`] = `<ul />`;

exports[`<TodoList /> renders with todos 1`] = `
<ul>
  <li
    onClick={[Function anonymous]}
    style={
      Object {
        "textDecoration": "none"
      }
    }>
    item a
  </li>
  <li
    onClick={[Function anonymous]}
    style={
      Object {
        "textDecoration": "line-through"
      }
    }>
    item b
  </li>
</ul>
`;
